### PR TITLE
Working on CLI commands

### PR DESCRIPTION
**Pull request by daddou-ma**
Before Run :
npm install 
npm install -g

Commands : jokes-cli params

Params 
 - add (for future xD)
 - all
 - count 
 - random
 - search regex
 - filter-by-lang lang
 - filter-by-tag tag1 tag2 .....tagn

### CODE REVIEW

File: jokes-cli.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+#! /usr/bin/env node
+const jokes      = require('./jokes.js')
```
Comment (houssem-yahiaoui) [inline jokes-cli.js]: What would be intresting is using the already present module as dependency, the stand alone jokes.dz won't be useful then, that's why i used it as dependency over my package.json file. so please change that :)

**Review by houssem-yahiaoui — CHANGES_REQUESTED** [review]

File: jokes.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const jokes = require('./jokes.json')
+
+
+/** 
+ * Number of total jokes available 
+ * @return {Number}
+ */
+const count = () => {
+  jokes.length
```
Comment (houssem-yahiaoui) [inline jokes.js]: Returns undefined, simply add the return before jokes.length

File: jokes-cli.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    if (arguments.length == 0) {
+        printHelp()
+        process.exit()
+    }
+
+    /** Get Command from the first argument **/
+    switch (arguments[0]) {
+        case "add":
+            // TODO : add joke (after add firebase)
+            break
```
Comment (youknowriad) [inline jokes-cli.js:22]: I think leaving unused code has not added value, I'd drop this case. If we ever implement adding jokes, we'll add it

File: jokes-cli.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            search(arguments[1])
+            break
+        case "filter-by-lang":
+            filterByLang(arguments[1])
+            break
+        case "filter-by-tag":
+           filterByTag(arguments.slice(1))
+            break
+        default:
+            // TODO : Unknown Arguments Error Msg
```
Comment (youknowriad) [inline jokes-cli.js]: Why todo, just a `throw new Error('some message')`

**Review by houssem-yahiaoui — DISMISSED** [review]

File: jokes-cli.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            message : 'image (url) : '  
+        },{
+            type    : 'input',
+            name    : 'tags',
+            message : 'tags : '  
+
+    }).then(function (answers) {
+        console.log('your answer is : ' + answers.joke)
+    })    
+}
```
Comment (youknowriad) [inline jokes-cli.js]: Same this function is not used yet I guess, I'll drop it for now

File: jokes.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    "img": null,
+    "tags": [ "زواج" ]
+  },
+  {
+    "lang": "ar",
+    "joke": null,
+    "img": "https://scontent-mrs1-1.xx.fbcdn.net/v/t1.0-9/15941503_1856068318002239_6728787835084139089_n.jpg?oh=9fdead02d71b0b87d7c7f2db0c1af2a4&oe=5919D061",
+    "tags": [ "conjugaison", "fr" ]
+  }
+]
```
Comment (youknowriad) [inline jokes.json]: I'll drop those files and use the external dependency instead as @houssem-yahiaoui mentionned

**Review by youknowriad — CHANGES_REQUESTED** [review]
This is great! thanks for this PR. Though, I left some small notes ;)

File: package.json [hunk 7]
```diff
@@ -15,9 +12,16 @@
...
+    "bin": {
+        "jokes-cli": "./jokes-cli.js"
+    },
+    "scripts": {
+        "jokes-cli": "./jokes-cli.js",
+        "start": "node app.js"
+    },
     "dependencies": {
         "commander": "^2.9.0",
         "figlet": "^1.2.0",
```
Comment (youknowriad) [inline package.json:27]: I know, you did not add those dependencies, but I think they are not used, are they?

Comment (youknowriad) [reply package.json:27]: I'm talking about commander and figlet

Comment (daddou-ma) [reply package.json:27]: Yes, ifound them in the first package.json but they are not used

Comment (youknowriad) [reply package.json:27]: Would you mind dropping them, though, I'm ok to leave this for a future PR.

Comment (youknowriad) [reply package.json:27]: You know what just leave them, we'll drop these after.

**Review by youknowriad — APPROVED** [review]

**Comment by youknowriad** [comment]
Tks @Mohamed-El-Amine-D you did a great job with this PR :)

File: package.json (hunk 7 above)
Comment (daddou-ma) [reply package.json:27]: okey ^^

**Comment by daddou-ma** [comment]
Thanks @youknowriad @houssem-yahiaoui  for your help and your feedbacks ^^
i really appreciate that :D

**Comment by youknowriad** [comment]
@Mohamed-El-Amine-D now we have to fill jokes.dz with jokes hahaha

**Comment by daddou-ma** [comment]
I'am Really a bad joker :o  xD, I let that task to an expert Lool

**Pull request merged**